### DISCUSSION TITLE
Split an object into seperate views

### DISCUSSION THREAD

**Discussion by jonocodes** [aminalaee/sqladmin#772] [General]
I have a table in my db called Users that has a combination of user data (name, email, etc) and user settings.

I would like to somehow show these as separate things. They can be different views/pages (perhaps User and UserSettings), or on the same page but in some kind of grouping.

Is there a recommended way of doing this without a lot of custom templates/forms?

I tried to create two different views that use the same table and then whitelisted fields there, but they both ended up going to the page for user listing.

**Comment by jacobgarder** [comment]
I think you are experiencing the same thing as discussed in issue #681. That the URL's is wrong when creating multiple views addressing the same model.

**Reply by jonocodes** [reply]
Thanks. I ended up creating 2 different orm models and pointed them at the same db table. This way I can control what fields are visible and how they are sorted at the model level instead on in admin.

**Reply by hasansezertasan** [reply]
> Thanks. I ended up creating 2 different orm models and pointed them at the same db table. This way I can control what fields are visible and how they are sorted at the model level instead on in admin.

I believe you can use Polymorphic Association Objects instead of creating OTO relationships between two different models.

**Reply by jonocodes** [reply]
Thanks. I am not sure I could go polymorphic here since the User table and model definition I cant change. I am trying to make 2 separate models in admin that dont have anything in common besides their id, so it wouldnt make much sense for them to build off a common base. And all this data belongs in a single table.

**Reply by hasansezertasan** [reply]
> Thanks. I am not sure I could go polymorphic here since the User table and model definition I cant change. I am trying to make 2 separate models in admin that dont have anything in common besides their id, so it wouldnt make much sense for them to build off a common base. And all this data belongs in a single table.

I see, so you have to use one table and you want to create multiple sqlalchemy models based on this table.

And finally, you want to create model views for each model based on the previous table?

**Comment by jonocodes** [comment]
> I see, so you have to use one table and you want to create multiple sqlalchemy models based on this table.

Yes.

> And finally, you want to create model views for each model based on the previous table?

Yes.
